File: edu-platform-master/edu-platform-master/src/main/java/edu/platform/domains/learner/service/LearnerService.java
package edu.platform.domains.learner.service;

import edu.platform.domains.learner.model.request.*;
import edu.platform.domains.learner.model.response.LearnerGetListResponse;
import edu.platform.domains.learner.model.response.LearnerResponse;

public interface LearnerService {
    LearnerResponse save(LearnerSaveRequest request);

    void active(Long learnerId, LearnerActiveRequest request);

    LearnerResponse auth(LearnerAuthRequest request);

    void mappingAdultAccount(LearnerMappingRequest request);

    void confirmMappingAdultAccount(LearnerConfirmMappingRequest request);

    LearnerGetListResponse getList(LearnerGetListRequest request);
}

File: edu-platform.admin-master/edu-platform.admin-master/src/services/learner/learner.service.ts
import http from "./learner.api";
import {LearnerGetListResponse} from "./response/LearnerGetListResponse";
import {LearnerGetListRequest} from "./request/LearnerGetListRequest";

const getLearners = (request: LearnerGetListRequest) => {
  return http.get<LearnerGetListResponse>(`/learners`, {
    params: request
  });
}

export const LearnerService = {
  getLearners
}